### ISSUE TITLE
Minion stops responding to commands after highstate with bad indentation in state

### ISSUE THREAD

**Issue by lucacorti** [saltstack/salt#22366]
After running:

```salt '*' state highstate test=True```

on the master, the minion stopped responding to any command until salt-minion was killed and restarted.

In the minion log:

```
 [salt.state                                  ][CRITICAL] Rendering SLS webserver failed, render error: while parsing a block mapping
  in "<unicode string>", line 1, column 1:
    webserver:
    ^
expected <block end>, but found '<block mapping start>'
  in "<unicode string>", line 5, column 3:
      user.present:
      ^
```

``` 
% salt-master --versions-report
           Salt: 2014.7.3
         Python: 2.7.9 (default, Dec 25 2014, 01:19:43)
         Jinja2: 2.7.3
       M2Crypto: 0.22
 msgpack-python: 0.4.2
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
```

```
% salt-minion --versions-report
           Salt: 2014.7.4
         Python: 2.7.3 (default, Mar 13 2014, 11:03:55)
         Jinja2: 2.7.3
       M2Crypto: 0.22
 msgpack-python: 0.4.6
   msgpack-pure: Not Installed
       pycrypto: 2.6.1
        libnacl: Not Installed
         PyYAML: 3.11
          ioflo: Not Installed
          PyZMQ: 14.5.0
           RAET: Not Installed
            ZMQ: 4.0.5
           Mako: Not Installed
```

**Comment by cachedout** [comment]
Hi @lucacorti.

I could not reproduce this at all. Could you provide a sample state which can reliably illustrate this error? Thanks!

**Comment by lucacorti** [comment]
I did some debugging. This occurs when refreshing pillar data on the minion.

I'm using `ext_pillar` with `gitfs`, and have a post-receive hook triggering `saltutil.refresh_pillar`.
When this happens, the minion stops respondig even to `test.ping`.

This is all I see in the debug log:

```
[INFO    ] User luca Executing command saltutil.refresh_pillar with jid 20150406230348459703
[DEBUG   ] Command details {'tgt_type': 'glob', 'jid': '20150406230348459703', 'tgt': '*', 'ret': '', 'user': 'luca', 'arg': [], 'fun': 'saltutil.refresh_pillar'}
[INFO    ] Starting a new job with PID 5777
[DEBUG   ] SaltEvent PUB socket URI: ipc:///var/run/salt/minion/minion_event_55ffc3a262_pub.ipc
[DEBUG   ] SaltEvent PULL socket URI: ipc:///var/run/salt/minion/minion_event_55ffc3a262_pull.ipc
[DEBUG   ] Sending event - data = {'_stamp': '2015-04-06T21:03:48.531215'}
[DEBUG   ] Handling event 'pillar_refresh||||||\x81\xa6_stamp\xba2015-04-06T21:03:48.531215'
[INFO    ] Returning information for job: 20150406230348459703
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
```

I tried stracing the process, it is stuck in a POLL.

**Comment by cachedout** [comment]
I think this same issue might be reported elsewhere. I know there are others who have reported it. Let me see if I can find it.

**Comment by cachedout** [comment]
We have this one: https://github.com/saltstack/salt/issues/20976
And this one: https://github.com/saltstack/salt/issues/19708

And this one: https://github.com/saltstack/salt/issues/19748

Any of those seem like they could be what you're seeing?

**Comment by lucacorti** [comment]
I'm not using multi-master, so #20976 seems a good candidate, the master is behind a pf firewall.
Another minion connected to the same master does not suffer from this though.

**Comment by lucacorti** [comment]
If the patch is safe to apply to 2014.7 I'll test this out, else I'm waiting for 2015.2.

**Comment by cachedout** [comment]
I have not personally tested that patch against 2014.7 but I see no reason offhand why it should not work.

**Comment by lucacorti** [comment]
I have adapted the patch and so far everithing seems working fine.

Thank you for the support. Any chance this will be backported to 2014.7?

**Comment by basepi** [comment]
I've marked it for backporting. I'm going to go ahead and close this, since that fix appears to be working for you. Thanks for the update!

**Comment by rallytime** [comment]
@lucacorti I have back-ported this fix to the 2014.7 branch in #22416.

**Comment by lucacorti** [comment]
@rallytime great!

**Comment by igorwidlinski** [comment]
The tcp keepalives do not work in 2014.7.5 .The backport missed line (original https://gist.github.com/kriberg/707df0bfec982256abe8 ) :
-            ZeroMQChannel.sreq_cache[key] = salt.payload.SREQ(self.master_uri)
+            ZeroMQChannel.sreq_cache[key] = salt.payload.SREQ(self.master_uri, opts=self.opts)

tested with above line and minions send tcp keep alives as supposed to.

**Comment by rallytime** [comment]
Good catch @lucacorti. This has been fixed at the head of the 2014.7 branch already with #23258. However, is should be noted that we won't be releasing from the 2014.7 branch again since 2015.5.0 is out. (The 2015.5 branch has the correct lines...I double checked.)
